back/app/services/admin_publications_filtering_service.rb

```diff
@@ -20,23 +20,23 @@ def for_homepage_filter(scope)
     scope.where(publication_type: Project.name)
   end
 
-  # This filter removes AdminPublications that represent folders which contain *only* projects which should not be visible to the current user.
-  # Here we are concerned with 'visibility' in terms of the Project.visible_to attribute, which can have one of 3 values: public, groups or admins.
-  add_filter('remove_not_allowed_parents') do |visible_publications, options|
-    next visible_publications unless ['true', true, '1'].include? options[:remove_not_allowed_parents]
-
-    public_project_ids          = Project.publicly_visible.ids
-    public_project_publications = AdminPublication.includes(:parent).where(publication_id: public_project_ids).where.not(parent_id: nil)
-
-    visible_or_public_publication_ids = visible_publications.map(&:parent_id).concat(public_project_publications.map(&:parent_id)).compact.uniq
```

Comment:
What about folders that have a project which is not public, but still visible to this user? Previously, they were included by `visible_publications.map(&:parent_id)`, but now they are filtered out. Should we still have `visible_publications.map(&:parent_id)`? Maybe even `visible_publications.not_draft.map(&:parent_id)`?

Reply:
I mean what about folders that have a project which is not public **and not draft**, but still visible to this user?

Maybe we could also use `scope.not_draft.ids.uniq` instead of `Project.publicly_visible.not_draft.ids.uniq`?

Reply:
Good point. That needs to be covered indeed. (And it means the spec is lacking because I didn't get any red tests? 🤔)

But, I think I might do this in a different ticket if we decide to do this. I'll revert it to the old behavior for now. I feel like I'm walking on thin ice.

Reply:
> And it means the spec is lacking because I didn't get any red tests?

> I feel like I'm walking on thin ice.

I agree with both.